src/gui/DetailsWidget.cpp

```diff
@@ -32,325 +32,291 @@
-        m_ui->tabWidget->insertTab(static_cast<int>(AttachmentsTab), m_attachmentsTabWidget, tr("Attachments"));
-        m_ui->tabWidget->insertTab(static_cast<int>(AutotypeTab), m_autotypeTabWidget, tr("Autotype"));
+void DetailsWidget::setGroup(Group* selectedGroup)
+{
+    if (!selectedGroup) {
+        hide();
+        return;
     }
 
-    m_ui->tabWidget->setTabEnabled(AttributesTab, false);
-    m_ui->tabWidget->setTabEnabled(NotesTab, false);
-    m_ui->tabWidget->setTabEnabled(AutotypeTab, false);
+    m_currentGroup = selectedGroup;
+    updateGroupHeaderLine();
+    updateGroupGeneralTab();
+    updateGroupNotesTab();
+
+    setVisible(!config()->get("GUI/HideDetailsView").toBool());
 
-    m_ui->totpButton->hide();
-    m_ui->totpWidget->hide();
-    m_ui->totpButton->setChecked(false);
+    m_ui->stackedWidget->setCurrentWidget(m_ui->pageGroup);
+    const int tabIndex = m_ui->groupTabWidget->isTabEnabled(m_selectedTabGroup) ? m_selectedTabGroup
+                                                                                : GeneralTabIndex;
+    Q_ASSERT(m_ui->groupTabWidget->isTabEnabled(GeneralTabIndex));
+    m_ui->groupTabWidget->setCurrentIndex(tabIndex);
+}
 
-    auto icon = m_currentEntry->iconPixmap();
-    if (icon.width() > 16 || icon.height() > 16) {
-        icon = icon.scaled(16, 16);
+void DetailsWidget::setDatabaseMode(DatabaseWidget::Mode mode)
+{
+    m_locked = mode == DatabaseWidget::LockedMode;
+    if (m_locked) {
+        return;
     }
-    m_ui->entryIcon->setPixmap(icon);
-
-    QString title = QString(" / ");
-    Group* entry_group = m_currentEntry->group();
-    if (entry_group) {
-        QStringList hierarchy = entry_group->hierarchy();
-        hierarchy.removeFirst();
-        title += hierarchy.join(" / ");
-        if (hierarchy.size() > 0) {
-            title += " / ";
+
+    if (mode == DatabaseWidget::ViewMode) {
+        if (m_ui->stackedWidget->currentWidget() == m_ui->pageGroup) {
+            setGroup(m_currentGroup);
+        } else {
+            setEntry(m_currentEntry);
         }
     }
-    title.append(m_currentEntry->resolveMultiplePlaceholders(m_currentEntry->title()));
-    m_ui->titleLabel->setText(title);
+}
+
+void DetailsWidget::updateEntryHeaderLine()
+{
+    Q_ASSERT(m_currentEntry);
+    const QString title = m_currentEntry->resolveMultiplePlaceholders(m_currentEntry->title());
+    m_ui->entryTitleLabel->setText(hierarchy(m_currentEntry->group(), title));
+    m_ui->entryIcon->setPixmap(preparePixmap(m_currentEntry->iconPixmap(), 16));
+}
+
+void DetailsWidget::updateEntryTotp()
+{
+    Q_ASSERT(m_currentEntry);
+    const bool hasTotp = m_currentEntry->hasTotp();
+    m_ui->entryTotpButton->setVisible(hasTotp);
+    m_ui->entryTotpWidget->hide();
+    m_ui->entryTotpButton->setChecked(false);
+
+    if (hasTotp) {
+        m_step = m_currentEntry->totpStep();
+        updateTotp();
+        m_totpTimer->start(m_step * 10);
```

Comment:
`m_timer->start(m_step * 1000);` (As fixed in #1414).

Reply:
`m_timer->start(m_step * 1000);`
@adolfogc @TheZ3ro @phoerious 
As for me, there's a bug. If we select an element at the twentieth second of the current round (the whole round takes for example 30 seconds), then first ten seconds will be correct and next twentieth not.

Reply:
Then you should use that as an offset. But letting the timer time out every few milliseconds is just a waste of processing power.

Reply:
I will try to fix it tomorrow

Reply:
I'd also be okay with running it unconditionally every second, but m_step * 10 makes no sense.

Reply:
Done. I set the timer for one second